gui/executionGraphGui/client/apps/eg/src/app/model/Connection.ts

```diff
@@ -54,14 +54,14 @@ export function isInvalid(source: Socket, target: Socket): Invalidity {
   /** no 1-length-cycles, more elaborate cycle-detection in the backend */
   result |= source.parentId !== target.parentId ? Invalidity.Valid : Invalidity.Invalid | Invalidity.OneLengthCycle;
   /** correct type */
-  result |= source.typeIndex.equals(target.typeIndex) ? Invalidity.Valid : Invalidity.Invalid | Invalidity.TypeMismatch;
+  result |= source.typeIndex === target.typeIndex ? Invalidity.Valid : Invalidity.Invalid | Invalidity.TypeMismatch;
   return result;
 }
 
 /**
  * Create a connection.
  */
-export function create(source: Socket, target: Socket, validate: boolean = true): Connection {
+export function createConnection(source: Socket, target: Socket, validate: boolean = true): Connection {
```

Comment:
I'd prefer 2 methods instead of a bool flag for turning validation on/off

Reply:
if keyword arguments would be on, it would be ok, but jeah maybe better